### DISCUSSION TITLE
Where to get the CoreWindow?

### DISCUSSION THREAD

**Discussion by TopperDEL** [unoplatform/uno#11569] [Q&A] [answered]
I need the CoreWindow to get the GetKeyState()-Method - but all attempts to get the current CoreWindow fail. If I understand correctly I should get it by calling:

`CoreWindow.GetForCurrentThread()`

But it is always null, even on the UI-thread. Any hints?

**Comment by FrozDark** [comment, marked as answer]
Seems like you are referencing to WinUI 3. The CoreWindow on Desktop is deprecated

References:
https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.window.corewindow?view=windows-app-sdk-1.2
https://github.com/microsoft/microsoft-ui-xaml/issues/4225

For the GetKeyState rather use https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.input.inputkeyboardsource.getkeystateforcurrentthread

**Reply by TopperDEL** [reply]
Thanks! That was what I was looking for!
